### PR TITLE
[FRD-604] Gnosis Safe, improved support

### PR DESCRIPTION
**Pull request by iower**

### CODE REVIEW

File: src/context/WalletProvider.tsx [hunk 1]
```diff
@@ -511,10 +516,34 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
...
+          /*
+            Gnosis Safe cannot immediately return the transaction by design.
+            The multisig signature will be made later.
+            It remains only to wait for the appearance of a new transaction from the sender's address.
+          */
+
+          // no await - no response expected
+          signer?.sendTransaction(transaction)
+
+          const waitForGnosisSafeTxHash: () => Promise<string> = () => {
```
Comment (insulineru) [inline src/context/WalletProvider.tsx]: it would be better to move from main code

File: src/context/WalletProvider.tsx [hunk 2]
```diff
@@ -511,10 +516,34 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
...
+            return new Promise((resolve, reject) => {
+              state.provider!.on({
```
Comment (insulineru) [inline src/context/WalletProvider.tsx]: wen `.off` 🧐

**Review by insulineru — COMMENTED** [review]
Your solution is cool ✌🏻

I would add some checks to it (e.g. that `event.to === tx.to`) and put it in a separate utility so as not to stretch the provider code.

File: src/context/WalletProvider.tsx (hunk 1 above)
Comment (iower) [reply src/context/WalletProvider.tsx]: Ok, moved

File: src/context/WalletProvider.tsx (hunk 2 above)
Comment (iower) [reply src/context/WalletProvider.tsx]: Already! :smile:

**Review by insulineru — APPROVED** [review]

**Pull request merged**